src/main/java/pl/cleankod/exchange/core/domain/Money.java

```diff
@@ -20,7 +20,7 @@ public static Money of(String amount, String currency) {
         return new Money(new BigDecimal(amount), Currency.getInstance(currency));
     }
 
-    public Money convert(CurrencyConversionService currencyConverter, Currency targetCurrency) {
-        return currencyConverter.convert(this, targetCurrency);
+    public Money convert(CurrencyConversionService currencyConverter, Currency targetCurrency, String correlationId) {
```

Comment:
Why did you 'bleed' the `correctionId` parameter into the core domain of the application?

Can you explain / do you know why this is a bad idea?

Do you know another way to approach this implementation that does not pollute the domain objects with the `correlationId`?